### PR TITLE
Exclude some files and directories by default, add flag to include them

### PR DESCRIPTION
**Pull request by igrr**
By default, don't add the following to the archive:

- .DS_Store files
- .git directories

Add new flag, -a/--all-files, which forces such files/directories to be
included anyway.

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -181,6 +188,21 @@ int addFiles(const char* dirname, const char* subPath) {
                 continue;
             }
 
+            if (!s_addAllFiles) {
+                bool skip = false;
+                size_t ignored_file_names_count = sizeof(ignored_file_names) / sizeof(ignored_file_names[0]);
+                for (size_t i = 0; i < ignored_file_names_count; ++i) {
+                    if (strcmp(ent->d_name, ignored_file_names[i]) == 0) {
```
Comment (me-no-dev) [inline main.cpp]: maybe use strncmp? for files like .gitignore and such?

Comment (igrr) [reply main.cpp]: I can add .gitignore and maybe .gitmodules into the excluded list, i think.

Comment (me-no-dev) [reply main.cpp]: either way will work I think :)

Comment (me-no-dev) [reply main.cpp]: or maybe we add option so the user can exclude whatever they want  (including wildcard) or would that be too much?

Comment (igrr) [reply main.cpp]: It could be a nice feature, but i will leave the implementation as an exercise to anyone who actually needs it.

Comment (igrr) [reply main.cpp]: Done!

Comment (me-no-dev) [reply main.cpp]: good enough for me :)

**Pull request merged**